### PR TITLE
mask for cpf_cnpj

### PR DESCRIPTION
**Pull request by andrersp**
Apply mask in cpf or cnpj if it is valid

### CODE REVIEW

File: pycpfcnpj/mask.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+from . import cpf
+from . import cnpj
```
Comment (matheuscas) [inline pycpfcnpj/mask.py:3]: Avoid relative imports. Use absolute imports.

File: tests/format_tests.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import unittest
+from pycpfcnpj import cpf, cnpj, mask
+
+
+class FORMATTests(unittest.TestCase):
+    """docstring for CPFTests"""
```
Comment (matheuscas) [inline tests/format_tests.py:6]: Remove it

File: tests/format_tests.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import unittest
+from pycpfcnpj import cpf, cnpj, mask
+
+
+class FORMATTests(unittest.TestCase):
```
Comment (matheuscas) [inline tests/format_tests.py:5]: Use PEP8 naming format. FormatTests, for instance. However, it's not a clear naming. What do you think about MaskingTests?

File: pycpfcnpj/mask.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+
+def mask_cpf_cnpj(cpf_cnpj: str):
+    cpf_cnpj = "".join([x for x in f'{cpf_cnpj}'])
```
Comment (matheuscas) [inline pycpfcnpj/mask.py:7]: I didn't get it why are you doing this.

File: pycpfcnpj/mask.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+    if cpf.validate(cpf_cnpj):
+        return f'{cpf_cnpj[:3]}.{cpf_cnpj[3:6]}.{cpf_cnpj[6:9]}-{cpf_cnpj[9:]}'
+    elif cnpj.validate(cpf_cnpj):
+        return f'{cpf_cnpj[:2]}.{cpf_cnpj[2:5]}.{cpf_cnpj[5:8]}/{cpf_cnpj[8:12]}-{cpf_cnpj[12:14]}'
+    else:
```
Comment (matheuscas) [inline pycpfcnpj/mask.py:13]: I think if it's not valid, it should raise an exception.

File: tests/format_tests.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    def setUp(self):
+        self.valid_cpf = '11144477735'
+        self.invalid_cpf = '11144477736'
+        self.invalid_cpf_str = '1234'
+        self.valid_cnpj = '11444777000161'
+        self.invalid_cnpj = '11444777000162'
+        self.invalid_cnpj_str = '1234'
+
+    def test_mask_valid_cpf(self):
+        self.assertTrue(cpf.validate(mask.mask_cpf_cnpj(self.valid_cpf)))
```
Comment (matheuscas) [inline tests/format_tests.py:17]: You already know that's a valid cpf. I think you should test whether or not the masked was properly applied. This comment applies for the tests below, except for the invalids that should check the exception raised.

**Review by matheuscas — COMMENTED** [review]
General comments:
- You should propose a documentation change as well in README about this new feature. 
- You must bump the version of the package in setup.py and in the CHANGELOG
- Change the CHANGELOG with you changes. 

Thanks for the contribution. :)

File: pycpfcnpj/mask.py (hunk 4 above)
Comment (andrersp) [reply pycpfcnpj/mask.py:7]: In some cases I get an integer with the cpf/cnpj number. That's why I do this conversion to string.

File: pycpfcnpj/mask.py (hunk 5 above)
Comment (andrersp) [reply pycpfcnpj/mask.py:13]: I have a case of consultation with the federal revenue base in which the 3 first and 2 last numbers of the cpf are hidden. Oh I leave this mask and not an exception

File: pycpfcnpj/mask.py (hunk 4 above)
Comment (matheuscas) [reply pycpfcnpj/mask.py:7]: You should raise an exception, then. See, you annotated that `cpf_cnpj` is a string, but still are doing an operation in the case that is not, you know what I mean?

File: pycpfcnpj/mask.py (hunk 5 above)
Comment (matheuscas) [reply pycpfcnpj/mask.py:13]: So your code must handle that, not the mask function. The mask function should have only and only one responsibility.

File: pycpfcnpj/mask.py (hunk 1 above)
Comment (matheuscas) [reply pycpfcnpj/mask.py:3]: Ps.: I know that the project has a lot of them, but at that time I wasn't aware that's a bad practice. ;) 
That's the beauty of refactoring. :)

**Comment by matheuscas** [comment]
No activity.

**Pull request closed without merging**